R/oauth-token.R

```diff
@@ -78,10 +78,13 @@ token_has_expired <- function(token, delay = 5) {
 }
 
 token_refresh <- function(client, refresh_token, scope = NULL, token_params = list()) {
-  oauth_client_get_token(client,
+  out <- oauth_client_get_token(
+    client,
     grant_type = "refresh_token",
     refresh_token = refresh_token,
     scope = scope,
     !!!token_params
   )
+  out$refresh_token <- out$refresh_token %||% refresh_token
```

Comment:
My understanding of the overall design is still tenuous enough that I'm not 100% sure of where to do this, but I think this is the right place.

For Google, the response for a token refresh looks like this:

```
{
  "access_token": "1/fFAGRNJru1FTz70BzhT3Zg",
  "expires_in": 3920,
  "scope": "https://www.googleapis.com/auth/drive.metadata.readonly",
  "token_type": "Bearer"
}
```

I confirmed this personally and it's also documented at <https://developers.google.com/identity/protocols/oauth2/native-app#offline>.

The main thing to notice is the LACK of the refresh token. So I think we need to proactively preserve it.